src/cascadia/TerminalApp/App.cpp

```diff
@@ -646,91 +289,30 @@ namespace winrt::TerminalApp::implementation
         // Here, we successfully reloaded the settings, and created a new
         // TerminalSettings object.
 
-        // Re-wire the keybindings to their handlers, as we'll have created a
-        // new AppKeyBindings object.
-        _HookupKeyBindings(_settings->GetKeybindings());
-
-        // Refresh UI elements
-
-        auto profiles = _settings->GetProfiles();
-        for (auto& profile : profiles)
-        {
-            const GUID profileGuid = profile.GetGuid();
-            TerminalSettings settings = _settings->MakeSettings(profileGuid);
-
-            for (auto& tab : _tabs)
-            {
-                // Attempt to reload the settings of any panes with this profile
-                tab->UpdateSettings(settings, profileGuid);
-            }
-        }
-
-        // Update the icon of the tab for the currently focused profile in that tab.
-        for (auto& tab : _tabs)
-        {
-            _UpdateTabIcon(tab);
-            _UpdateTitle(tab);
-        }
+        // Inform Page to update the UI
+        terminalPage->RefreshUIAfterSettingsReloaded();

```

Comment:
>RefreshUIAfterSettingsReloaded [](start = 22, length = 30)

Is this needed?  Should this be an automatic side-effect of the SetSettings() call above?

Reply:
I think this method is needed. Changing the settings will not trigger automatic UI updates. Originally, App::_ReloadSettings goes through each tab and does the update.

Reply:
I dunno, I kinda agree that it should maybe be included as a side-effect of `SetSettings`. While the first call to `SetSettings` doesn't really need to refresh the UI (since the UI hasn't even been `Create`d yet), I'd think that _every_ subsequent call to `SetSettings` would _definitely_ want the UI to refresh